File: leetcode/0326_power_of_three.cpp
// Copyright (c) 2018 Jacopo Notarstefano

#include <cassert>

using namespace std;

class Solution {
 public:
    bool isPowerOfThree(int n) {
        return
            n == 1 ||
            n == 3 ||
            n == 9 ||
            n == 27 ||
            n == 81 ||
            n == 243 ||
            n == 729 ||
            n == 2187 ||
            n == 6561 ||
            n == 19683 ||
            n == 59049 ||
            n == 177147 ||
            n == 531441 ||
            n == 1594323 ||
            n == 4782969 ||
            n == 14348907 ||
            n == 43046721 ||
            n == 129140163 ||
            n == 387420489 ||
            n == 1162261467;
    }
};

int main() {
    auto solution = Solution();

    assert(solution.isPowerOfThree(27));
    assert(!solution.isPowerOfThree(0));
    assert(solution.isPowerOfThree(9));
    assert(!solution.isPowerOfThree(45));
}

File: leetcode/0326_power_of_three.py
# -*- coding: utf-8 -*-


class Solution(object):
    def isPowerOfThree(self, n):
        return (
            n == 1 or
            n == 3 or
            n == 9 or
            n == 27 or
            n == 81 or
            n == 243 or
            n == 729 or
            n == 2187 or
            n == 6561 or
            n == 19683 or
            n == 59049 or
            n == 177147 or
            n == 531441 or
            n == 1594323 or
            n == 4782969 or
            n == 14348907 or
            n == 43046721 or
            n == 129140163 or
            n == 387420489 or
            n == 1162261467
        )


if __name__ == '__main__':
    solution = Solution()

    assert solution.isPowerOfThree(27)
